Add changelog and upgrade guide for 2.235.4

MarkEWaite opened the pull request:
## Add changelog and upgrade guide for 2.235.4

Intentionally duplicated the 2.235.3 upgrade guide entry into 2.235.4 in hopes that administrators will read and heed what it says.  No need to duplicate it on later releases, but one duplication seems a small price to pay in hopes of reducing issue reports.

![Annotation 2020-08-11 104402](https://user-images.githubusercontent.com/156685/89925782-1ffbfb00-dbc1-11ea-948e-6da5845c2c04.png)


### Prior Changelog

![Annotation 2020-08-10 192921](https://user-images.githubusercontent.com/156685/89846943-d4553d00-db3f-11ea-9877-84f537b0b56f.png)

### Upgrade Guide

![screencapture-localhost-4242-doc-upgrade-guide-2-235-2020-08-11-10_44_20-edit](https://user-images.githubusercontent.com/156685/89925860-402bba00-dbc1-11ea-8d24-1d6e81030ae4.png)

daniel-beck commented on content/_data/upgrades/2-235-4.adoc:
```diff
@@ -0,0 +1,26 @@
+==== Repository Signing Key Update
```
This file is redundant as we have one page per LTS baseline, these are then just sections on that page. Would rather do perhaps a changelog banner than this.

MarkEWaite replied:
Thanks. Fixed in https://github.com/jenkins-infra/jenkins.io/pull/3586/commits/616a00dcf447c609eef3cec01c7363fb1c9cddc3

daniel-beck commented:
> Would it make sense to mention the new Windows installer as a major RFE? It may also require some upgrade guidelines.

Wasn't that part of .3?

oleg-nenashev commented:
> Wasn't that part of .3?

Nope, we skipped the windows installer there. It has not been updated retrospectively from what I see on http://mirrors.jenkins-ci.org/windows-stable/

![image](https://user-images.githubusercontent.com/3000480/89387626-0c551e00-d703-11ea-8b8c-be5cbde6fc0b.png)

MarkEWaite commented:
@oleg-nenashev I've added a first attempt at the Windows installer upgrade guide in https://github.com/jenkins-infra/jenkins.io/pull/3586/commits/8f18073a88039b9d65001ce53bb8f50c75e5586f .  Would love to have reviews from @slide , @kwhetstone , and @vsilverman .  There are many more details that could be provided.

slide commented on content/_data/upgrades/2-235-3.adoc:
```diff
@@ -24,3 +24,46 @@ Update Red Hat compatible operating systems (Red Hat Enterprise Linux, CentOS, F
+Beginning with Jenkins LTS release link:/changelog-stable/#v2.235.3[2.235.3], the same Windows installers is used in the stable release as has been used since April 2020 in the weekly release.
+The new Windows installer was announced in a link:/blog/2019/02/01/windows-installers/[blog post].
+
+Jenkins administrators on Windows may choose to continue using and upgrading their existing installation or they may choose to migrate to the new Windows installer.
```
```suggestion
Jenkins administrators on Windows may choose to continue using and upgrading their existing installation using the mechanism in the Jenkins UI, or they may choose to migrate to the new Windows installer.
```

slide commented on content/_data/upgrades/2-235-3.adoc:
```diff
@@ -24,3 +24,46 @@ Update Red Hat compatible operating systems (Red Hat Enterprise Linux, CentOS, F
+Administrators generally do not use the Jenkins MSI installer to perform a Jenkins upgrade because the upgrade is simple and easy from within Jenkins itself.
+
+Administrators that choose to continue using their existing installation will continue to be able to upgrade to new Jenkins versions.
+Settings in the Jenkins installation can be modified by editing the `jenkins.exe.config.xml` file and other configuration files.
```
```suggestion
Settings in the Jenkins installation can be modified by editing the `jenkins.exe.config` file and other configuration files.
```

slide commented on content/_data/upgrades/2-235-3.adoc:
```diff
@@ -24,3 +24,46 @@ Update Red Hat compatible operating systems (Red Hat Enterprise Linux, CentOS, F
+Administrators that choose to continue using their existing installation will continue to be able to upgrade to new Jenkins versions.
+Settings in the Jenkins installation can be modified by editing the `jenkins.exe.config.xml` file and other configuration files.
+
+In this configuration, Jenkins will continue to use the 32 bit Java Runtime Environment that was included with the previous installer.
```
```suggestion
In this configuration, Jenkins will continue to use the 32 bit Java Runtime Environment that was included with the previous installer, unless they manually modify the jenkins.exe.config to point to a 64 bit Java Runtime Environment.
```

vsilverman replied:
It may be useful referencing a valid 64 bit Java Runtime Environment configuration inside jenkins.exe.config file.
Content of jenkins.exe.config file, generated by Jenkins installer 2.235.2, doesn't contain explicit reference to 32 bit Java Runtime Environment. For this reason it's not obvious which fields should be modified for pointing to a 64 bit Java Runtime Environment.

slide commented on content/_data/upgrades/2-235-3.adoc:
```diff
@@ -24,3 +24,46 @@ Update Red Hat compatible operating systems (Red Hat Enterprise Linux, CentOS, F
+
+The new Windows installer provides the following additional features:
+
+* Choose the 64 bit Java 8 or 64 bit Java 11 installation to run Jenkins
```
```suggestion
* Choose a 64 bit Java 8 or 64 bit Java 11 installation to run Jenkins
```

MarkEWaite commented:
@slide are the most recent changes sufficient for you to approve this change?

@oleg-nenashev are the changes sufficient for your concerns as well?

daniel-beck commented:
> One of limitations I experienced with internal 2.235.2 upgrade - adding JDK 11 in Global Tool Config is not supported.
> The same limitation is present after internal Jenkins upgrade to 2.235.3.
> Does it make sense mentioning limitations of internal Jenkins upgrade?

Is that really _specific to_ that way to upgrade?

MarkEWaite commented:
> > One of limitations I experienced with internal 2.235.2 upgrade - adding JDK 11 in Global Tool Config is not supported.
> > The same limitation is present after internal Jenkins upgrade to 2.235.3.
> > Does it make sense mentioning limitations of internal Jenkins upgrade?
> 
> Is that really _specific to_ that way to upgrade?

I think we may be combining two unrelated things here.  The "Global Tool Config" page from within the "Manage Jenkins" page allows me to configure automatically installed JDK's for agents.  The Jenkins JDK installer plugin does not show Oracle JDK 11 as an available option on any of my Jenkins installations.  I thought that was intentional and somehow related to Oracle license terms for JDK 11.  The text on the Jenkins page says

> Oracle Java SE 11+ is not available for business, commercial or production use without a commercial license

After installing the AdoptOpenJDK plugin, I see "install from adoptopenjdk.net" as an option.  That allows me to install JDK 11 as provided by AdoptOpenJDK.

@vsilverman , have I understood what you were describing?  If not, could you describe in more detail?  Alternately, we could meet over Google Meet or Zoom so that I could see what you're seeing.

oleg-nenashev commented on content/_data/upgrades/2-235-3.adoc:
```diff
@@ -24,3 +24,46 @@ Update Red Hat compatible operating systems (Red Hat Enterprise Linux, CentOS, F
 
 See the link:/blog/2020/07/27/repository-signing-keys-changing/[blog post] for more details.
+
+==== New Windows Installer
```
I would rather move it to .4. Nobody reads changelogs retrospectively.

MarkEWaite replied:
Done in b90e4d48823ba4b8c66a2be24981cb260c38e980

daniel-beck commented:
> Windows packaging change is still missing

It's a banner with bold text though?

MarkEWaite commented:
@oleg-nenashev I've moved the new Windows installer from banner to major RFE in the changelog 15a1c7c7d852a1d09efc21a80a5031d3d9ed33a0 with intentionally bold text for the phrase "new Windows installer".  I'm hoping that increases the likelihood of reading the linked documents

I've moved the upgrade guide into 2.235.4 b90e4d48823ba4b8c66a2be24981cb260c38e980 so that users who click the 2.235.4 (rather than 2.235.x) will see the upgrade guide.

vsilverman commented:
@MarkEWaite, thanks for clarifying, indeed AdoptOpenJDK plugin removes those limitations. 

> On Aug 11, 2020, at 7:49 AM, Mark Waite <notifications@github.com> wrote:
> 
> 
> One of limitations I experienced with internal 2.235.2 upgrade - adding JDK 11 in Global Tool Config is not supported.
> The same limitation is present after internal Jenkins upgrade to 2.235.3.
> Does it make sense mentioning limitations of internal Jenkins upgrade?
> 
> Is that really specific to that way to upgrade?
> 
> I think we may be combining two unrelated things here. The "Global Tool Config" page from within the "Manage Jenkins" page allows me to configure automatically installed JDK's for agents. The Jenkins JDK installer plugin does not show Oracle JDK 11 as an available option on any of my Jenkins installations. I thought that was intentional and somehow related to Oracle license terms for JDK 11. The text on the Jenkins page says
> 
> Oracle Java SE 11+ is not available for business, commercial or production use without a commercial license
> 
> After installing the AdoptOpenJDK plugin, I see "install from adoptopenjdk.net" as an option. That allows me to install JDK 11 as provided by AdoptOpenJDK.
> 
> @vsilverman <https://github.com/vsilverman> , have I understood what you were describing? If not, could you describe in more detail? Alternately, we could meet over Google Meet or Zoom so that I could see what you're seeing.
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub <https://github.com/jenkins-infra/jenkins.io/pull/3586#issuecomment-671992788>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AAQPKTE2IOSBJHHTDF2FECDSAFLBBANCNFSM4PTQYICQ>.
>

daniel-beck commented:
I cannot wait any longer, so merging this now. Please refrain from merging any amendments before the releases are out this afternoon. Thanks!
